spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+                    "SELECT AlbumId, AlbumTitle, MarketingBudget FROM Albums@ "
+                    + "{FORCE_INDEX=AlbumsByAlbumTitle2} "
+                    + $"WHERE AlbumTitle >= '{startTitle}' "
+                    + $"AND AlbumTitle < '{endTitle}'");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]} "
+                        + $"MarketingBudget : {reader["MarketingBudget"]}");
+                    }
+                }
+            }
+            return;
+            // [END read_data_with_storing_index]
+        }
+
+        public static async Task QueryDataWithTransactionAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START read_only_transaction]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Gets a transaction object that captures the database state
+            // at a specific point in time.
+            using (TransactionScope scope = new TransactionScope(
+                TransactionScopeAsyncFlowOption.Enabled))
+            {
+                // Create connection to spanner.
+                using (var connection = new SpannerConnection(
+                    connectionString))
+                {
+                    var cmd = connection.CreateSelectCommand(
+                        "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                    // Read #1.
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Console.WriteLine(
+                                $"SingerId : {reader["SingerId"]}"
+                                + $" AlbumId : {reader["AlbumId"]}"
+                                + $" AlbumTitle : {reader["AlbumTitle"]}");
+                        }
+                    }
+                    // Read #2. Even if changes occur in-between the reads, 
+                    // the transaction ensures that Read #1 and Read #2 
+                    // return the same data.
+                    using (var reader = await cmd.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            Console.WriteLine(
+                                $"SingerId : {reader["SingerId"]}"
+                                + $" AlbumId : {reader["AlbumId"]}"
+                                + $" AlbumTitle : {reader["AlbumTitle"]}");
+                        }
+                    }
+                }
+                // Commit the transaction to Spanner table.

```

Comment:
I don't think the below complete() actually does anything on the backend since read only transactions dont need to be committed. So you can just omit this comment. It is kind of misleading.